snmp/datadog_checks/snmp/snmp.py

```diff
@@ -75,6 +76,8 @@ def __init__(self, *args, **kwargs):
 
         self._config = self._build_config(self.instance)
 
+        self._last_request_id = 0
```

Comment:
if `self._last_request_id` is attached to a specific check instance, multiple instances will have the same `self._last_request_id ` (and we won't be able to distinguish them) ?

Reply:
Yes, good point, switched to a global variable. (Sounds hacky, but I don't have a better solution? 🤷)

Reply:
1/ Actually, I was wrong about this point.

We can use both `check_id` + `fetch_id` to be sure it's about the same request.

But I think it will be more user friendly to have a `fetch_id` a string containing both check_id and instance specific fetch id.

1a/ Example: `<CHECK_ID>-<UNIQUE_COUNTER_PER_INSTANCE>` e.g. `a1b2c3-10`

1b/ To avoid maintaining a counter, we can `<CHECK_ID>-<6_OR_8_CHAR_ALPHANUM_RANDOM_UUID>`
e.g. `a1b2c3-x1y2z3`

2/ Using a global is worse since we might have race condition when changing the variable.

Reply:
@florimondmanca In the last commit 7b9b98f1deae020a4bbc056f111a5ebb12555f22 I think using a global `_LAST_FETCH_ID` might cause concurrency issue (The agent can run multiple checks instances at the same time and this might cause an issue when updating this global). I think it can be avoided by using instance level counter (like initial commit) and use `<CHECK_ID>-<UNIQUE_COUNTER_PER_INSTANCE>` as request id.